classes/class-pmpro-addons.php

```diff
@@ -907,9 +910,14 @@ private function send_ajax_result( $result ) {
 	 * @since 2.8.x
 	 *
 	 * @param string $slug The identifying slug for the addon (typically the directory name).
-	 * @return string $plugin_icon_src The src URL for the plugin icon.
+	 * @return string|false $plugin_icon_src The src URL for the plugin icon or false if PMPRO_DIR is not defined.
 	 */
 	public function get_addon_icon( $slug ) {
+		// If PMPRO_DIR is not defined, bail. This may happen in the Update Manager Add On.
+		if ( ! defined( 'PMPRO_DIR' ) ) {
+			return false;
```

Comment:
Instead of returning false, we could return the default in the get_addon_icon method. That way we don't have to check for the false return above.